### PR TITLE
chore(webapp): timeline chart, removed circles on dataPoints, added crosshair cursor

### PR DESCRIPTION
**Pull request by StasDachinsky**
## Brief
- https://github.com/pyroscope-io/pyroscope/issues/1400
## Changes
 - added crosshair cursor
 - removed circles in data points on line charts
 - changed lineWidth to line chart from 1 to 0.8
## Concerns
- I think that simply reducing the line width for line charts is not the best option, and we need to do this dynamically based on the lengths of dataPoints and Apps arrays. Reducing the width looks good for large time periods (and big data), but almost useless for small

https://user-images.githubusercontent.com/23450818/186914561-e0d707d9-6cb9-4f89-baec-28110d336d7b.mov

### CODE REVIEW

File: webapp/javascript/components/TimelineChart/TimelineChartSelection.ts [hunk 1]
```diff
@@ -116,8 +119,14 @@ const handleHeight = 22;
       return isLeftSelecting ? 'left' : isRightSelecting ? 'right' : null;
     }
 
-    function setCursor(type: string) {
-      $('canvas.flot-overlay').css('cursor', type);
+    function cursor() {
+      let current = '';
+      return function setCursor(type: string) {
```
Comment (dogfrogfog) [inline webapp/javascript/components/TimelineChart/TimelineChartSelection.ts:124]: can we use anonymous function here ?

File: webapp/javascript/components/TimelineChart/TimelineChartSelection.ts [hunk 2]
```diff
@@ -116,8 +117,14 @@ const handleHeight = 22;
       return isLeftSelecting ? 'left' : isRightSelecting ? 'right' : null;
     }
 
-    function setCursor(type: string) {
-      $('canvas.flot-overlay').css('cursor', type);
+    function cursor() {
```
Comment (pavelpashkovsky) [inline webapp/javascript/components/TimelineChart/TimelineChartSelection.ts:120]: function `cursor`? please add what does this function do to the name. setCursor? changeCursor?

File: webapp/javascript/components/TimelineChart/TimelineChartSelection.ts [hunk 3]
```diff
@@ -116,8 +117,14 @@ const handleHeight = 22;
...
+      let current = '';
+      return function (type: string) {
```
Comment (pavelpashkovsky) [inline webapp/javascript/components/TimelineChart/TimelineChartSelection.ts:122]: what the point of returning function? can you simplify it?

File: webapp/javascript/components/TimelineChart/TimelineChartSelection.ts [hunk 4]
```diff
@@ -389,6 +396,7 @@ const handleHeight = 22;
     plot.clearSelection = clearSelection;
     plot.setSelection = setSelection;
     plot.getSelection = getSelection;
+    plot.setCursor = cursor();
```
Comment (pavelpashkovsky) [inline webapp/javascript/components/TimelineChart/TimelineChartSelection.ts:399]: can you please make name of this function more informative cursor();

plot.setCursor = setCursor(); ?

**Review by pavelpashkovsky — CHANGES_REQUESTED** [review]

**Comment by pavelpashkovsky** [comment]
@StasDachinsky can you please take a look at 
pyroscope/webapp/javascript/components/TimelineChart/TimelineChartPlugin.ts

at `function bindEvents(plot, eventHolder)` you can add any handlers (or use existing) to it. might be helpful

**Review by eh-am — COMMENTED** [review]
I don't quite understand why do we need all these changes, if we already support it here https://github.com/pyroscope-io/pyroscope/blob/5788cf9943a579acd35de224537016488a98808f/webapp/sass/profile.scss#L110-L116

I feel like this implementation does more than the trivial impl above, can you just explain it what it does?

Also, if we are going with this route, we should remove that css I linked above then.

File: webapp/javascript/components/TimelineChart/TimelineChartPlugin.ts [hunk 5]
```diff
@@ -57,7 +58,17 @@ import { getUTCdate } from '@webapp/util/formatDate';
       }
     }
 
+    const setHoverCursor = (isHover: boolean) => {
+      if (this.isHover !== isHover) {
```
Comment (eh-am) [inline webapp/javascript/components/TimelineChart/TimelineChartPlugin.ts:62]:
```suggestion
      const hasChanged = this.isHover !== isHover;
      if (hasChanged) {
```

a nit, but I found these are clearer to read, any chance to reduce mental overhead is a good idea

File: webapp/javascript/components/TimelineChart/TimelineChartSelection.ts [hunk 6]
```diff
@@ -20,6 +20,7 @@ type PlotType = {
   clearSelection: (preventEvent: boolean) => void;
   setSelection: (ranges: any, preventEvent: any) => void;
   getSelection: () => {} | null;
+  setCursor: (cursor: string) => void;
```
Comment (eh-am) [inline webapp/javascript/components/TimelineChart/TimelineChartSelection.ts:23]:
```suggestion
  setCursor: (cursor: 'grab' | 'crosshair') => void;
```

If possible let's narrow the types as much as possible. By having just `string` thereotically a `setCursor("FOOOOOOO")` is valid, although it's obviously not true.

File: webapp/javascript/components/TimelineChart/TimelineChartSelection.ts [hunk 7]
```diff
@@ -116,8 +117,14 @@ const handleHeight = 22;
       return isLeftSelecting ? 'left' : isRightSelecting ? 'right' : null;
     }
 
-    function setCursor(type: string) {
-      $('canvas.flot-overlay').css('cursor', type);
+    function cursor() {
+      let current = '';
+      return function (type: string) {
+        if (type && current !== type) {
```
Comment (eh-am) [inline webapp/javascript/components/TimelineChart/TimelineChartSelection.ts:123]: Why do you have to check `type` is truthy? If it's always truthy we don't need to check it, and if it's not truthy then we need to indicate that in the types (ie `(type: string)` becomes `(type?: string)`)

File: webapp/javascript/components/TimelineChart/TimelineChartSelection.ts [hunk 8]
```diff
@@ -116,8 +117,14 @@ const handleHeight = 22;
...
+          current = type;
+          $('canvas.flot-overlay').css('cursor', type);
```
Comment (eh-am) [inline webapp/javascript/components/TimelineChart/TimelineChartSelection.ts:125]: What if there are 2 canvas in the same page?

File: webapp/javascript/components/TimelineChart/TimelineChartWrapper.tsx [hunk 9]
```diff
@@ -126,6 +127,7 @@ class TimelineChartWrapper extends React.Component<
       },
       points: {
         show: false,
+        symbol: () => {}, // function that draw points on the graph
```
Comment (eh-am) [inline webapp/javascript/components/TimelineChart/TimelineChartWrapper.tsx:130]: Nice!

File: webapp/javascript/components/TimelineChart/TimelineChartSelection.ts (hunk 8 above)
Comment (eh-am) [reply webapp/javascript/components/TimelineChart/TimelineChartSelection.ts:125]: I wonder if there isn't a way to get the actual element we care about cc @pavelpashkovsky

**Review by Rperry2174 — APPROVED** [review]
LGTM

**Pull request merged**